ADAL/src/ADAuthenticationContext.m

```diff
@@ -118,10 +131,12 @@ - (id)initWithAuthority:(NSString *)authority
     ADTokenCache* cache = [ADTokenCache new];
     [cache setDelegate:delegate];
     
-    return [self initWithAuthority:authority
+    self = [self initWithAuthority:authority
                  validateAuthority:validateAuthority
                         tokenCache:cache
                              error:error];
+    SAFE_ARC_RELEASE(cache);
+    return self;
```

Comment:
Sorry, wrong answer. In this case we didn't call alloc (someone else did) so we don't have a retain count to release or autorelease.